bootstrap/bootstrap.go

```diff
@@ -153,13 +169,45 @@ func Run(
 	})
 
 	// call individual bootstrap handlers.
+	startedSuccessfully := true
 	for i := range handlers {
 		if handlers[i](ctx, &wg, startupTimer, dic) == false {
 			cancel()
+			startedSuccessfully = false
 			break
 		}
 	}
 
+	return &wg, startedSuccessfully
+}
+
+// Run bootstraps an application.  It loads configuration and calls the provided list of handlers.  Any long-running
+// process should be spawned as a go routine in a handler.  Handlers are expected to return immediately.  Once all of
+// the handlers are called this function will wait for any go routines spawned inside the handlers to exit before
+// returning to the caller.  It is intended that the caller stop executing on the return of this function.
+func Run(
+	ctx context.Context,
+	cancel context.CancelFunc,
+	commonFlags commandline.CommonFlags,
+	serviceKey string,
+	configStem string,
+	serviceConfig interfaces.Configuration,
+	startupTimer startup.Timer,
+	dic *di.Container,
+	handlers []interfaces.BootstrapHandler) {
+
+	wg, _ := RunAndReturnWaitGroup(
```

Comment:
Shouldn't you check startedSuccessfully and call fatalError() if it didn't? How did we deal with error in bootstrap handlers before?

Reply:
Run() doesn't care about whether or not things started successfully.  

If they did, Run will wg.Wait() for sigterm to occur (which calls cancel, which cleanly stops all the goroutines, which ultimately causes wg.Wait() to return).  

If they didn't, the handler that failed will return false (which causes cancel to be called, which cleanly stops all the goroutines, which ultimately causes wg.Wait() to return).